### PR TITLE
[Cosmos] Use []json.RawMessage for cosmos list unmarshal to avoid MaxInt64 mishandling

### PR DESCRIPTION
**Pull request by xsuo2022**
Fixes https://github.com/Azure/azure-sdk-for-go/issues/22523

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/data/azcosmos/cosmos_query_response.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func newQueryResponse(resp *http.Response) (QueryItemsResponse, error) {
 }
 
 type queryServiceResponse struct {
-	Documents []any `json:"Documents,omitempty"`
+	Documents []json.RawMessage `json:"Documents,omitempty"`
```
Comment (ealsur) [inline sdk/data/azcosmos/cosmos_query_response.go:61]: Nice fix! Thank you for the tests too!

**Review by ealsur — APPROVED** [review]

File: sdk/data/azcosmos/cosmos_query_response.go (hunk 1 above)
Comment (ealsur) [reply sdk/data/azcosmos/cosmos_query_response.go:61]: @xsuo2022 I think you need to agree to the contribution policy based on the comments for this PR to merge

**Comment by xsuo2022** [comment]
@microsoft-github-policy-service agree company="Confluent"

File: sdk/data/azcosmos/cosmos_query_response.go (hunk 1 above)
Comment (xsuo2022) [reply sdk/data/azcosmos/cosmos_query_response.go:61]: @ealsur Just signed the CLA, thanks!

**Comment by ealsur** [comment]
@xsuo2022 at the moment we are working for GA release, which tentatively is end of march. I think the Azure SDKs release cycle is the first day of the next month.

**Pull request merged**